openhtf/__init__.py

```diff
@@ -217,8 +218,17 @@ def execute(self, test_start=None):
       self._test_desc.metadata['config'] = conf._asdict()
       self.last_run_time_millis = util.time_millis()
 
+      if not any((isinstance(test_start, PhaseDescriptor), test_start is None)):
```

Comment:
Something to consider, by checking for a `PhaseDescriptor` here, the start trigger phase *must* be decorated with something that does the wrapping.  This means that you can't use undecorated functions, which is inconsistent with test phases.